### PR TITLE
Update dependencies for MVO

### PR DESCRIPTION
**Pull request by cblecker**
- Updates velero to 1.5.1
- Updates openshift/k8s dependencies to 4.5/1.8
- Updates operator-sdk to v0.18.1
- Updates prometheus-operator to v0.42
- Copies `generateServiceMonitor` function from operator-sdk, as it's being removed in a future version

### CODE REVIEW

**Comment by cblecker** [comment]
/hold
for review

File: pkg/controller/velero/velero.go [hunk 1]
```diff
@@ -582,3 +581,46 @@ func credentialsRequestSpecEqual(x, y minterv1.CredentialsRequestSpec) (bool, er
 
 	return true, nil
 }
+
+// GenerateServiceMonitor generates a prometheus-operator ServiceMonitor object
+// based on the passed Service object.
+func generateServiceMonitor(s *corev1.Service) *monitoringv1.ServiceMonitor {
```
Comment (yithian) [inline pkg/controller/velero/velero.go:587]: if this is a function that we're going to own then we should have a unit test for it imo

**Review by yithian — CHANGES_REQUESTED** [review]
version updates all seem fine, i'd like to see a unit test for `generateServiceMonitor(...)` though

**Review by yithian — APPROVED** [review]

**Pull request merged**